src/vm/syncblk.cpp

```diff
@@ -2554,6 +2554,17 @@ inline void LogContention()
 #define LogContention()
 #endif
 
+LONGLONG ComputeElapsedTimeInNanosecond(LARGE_INTEGER startTime, LARGE_INTEGER endTime)
+{
+    static LARGE_INTEGER freq;
+    if (freq.QuadPart == 0)
+        QueryPerformanceFrequency(&freq);
+
+    const int NsPerSecond = 1000 * 1000 * 1000;
+    LONGLONG elapsedTicks = endTime.QuadPart - startTime.QuadPart;
+    return (elapsedTicks * NsPerSecond) / freq.QuadPart;
```

Comment:
It would probably make sense to compute and return a double value without int truncation

Reply:
you are right.